src/git_provider/mod.rs

```diff
@@ -47,11 +48,27 @@ pub trait GitProvider {
     fn set_token(&self, path: &PathBuf) -> Result<String>;
     fn get_token(&self) -> Result<String>;
 
+    fn set_title(&self, branch: &String) -> Result<String> {
+        let title = dialoguer::Input::<String>::with_theme(&ColorfulTheme::default())
+            .with_prompt("Pull request title:")
+            .default(branch.to_owned())
+            .interact()?;
+
+        Ok(title)
+    }
+
+    fn set_body(&self) -> Result<String> {
+        let body = dialoguer::Input::<String>::with_theme(&ColorfulTheme::default())
+            .with_prompt("Pull request body (press enter to leave it empty):")
+            .allow_empty(true)
+            .interact()?;
+        Ok(body)
+    }
```

Comment:
Don't declare these functions here, inside the trait. They are not related to the GitProvider trait. Please declare them outside the trait, as normal functions.